sw/device/lib/testing/test_framework/ottf_isrs.S

```diff
@@ -56,9 +56,13 @@ compute_mepc_on_synchronous_irq:
 .balign 4
 .type save_current_sp_to_tcb, @function
 save_current_sp_to_tcb:
-  la t0, kTestConfig  // Load the pointer to the test configuration struct.
-  lb t1, 0(t0)        // Load the first byte of the test_config struct, which
-                      // contains the `enable_concurrency` field to check.
+  la t0, kTestConfig       // Load the pointer to the test configuration struct.
+  beqz t0, .L_skip_sp_save // If the kTestConfig pointer is NULL, we are using
+                           // these ISRs without the OTTF, so no need to check
+                           // if concurrency is enabled.
+  lb t1, 0(t0)             // Load the first byte of the test_config struct,
+                           // which contains the `enable_concurrency` field to
+                           // check.
```

Comment:
I am wondering if these checks are needed at all? The reasoning is that it skips unnecessary stack pointer save/restore in the cases when there is only one task, but it does add extra logic that is executed in both cases (6 instructions on entry and exit when multitasking, 2 instructions when not), as well as a lot of comments.

I guess another argument would be that it actually would be more efficient, due to saving 4 cycles on every context switch - which would happen more frequently than other exceptions.

Just wanted to give you an alternative opinion, it is not a big deal in any case, and I see the appeal doing it the way you done it.

If you decide to go with my suggestion, I think dropping these two layers of branching (making `lw t2, pxCurrentTCB; sw sp, 0(t2)` unconditional, and the same for the exit routine), and making a short comment saying that stack pointer is saved/restored unconditionally - would be a reasonable alternative.

Please let me know if you think that I am missing something.

Reply:
The reason it is done this way, is to enable using the OTTF ISRs lib without the main OTTF lib (as in the case in the crt_test, and some other tests as well, see #9335). This allows us to only have to  maintain a single set of default ISRs/exception handlers for testing (see #9575). Basically we give test developers two options: 1) use just the default OTTF ISRs/exception handlers and let them implement their own `main()`, or 2) use the full blown OTTF (which includes the default ISRs lib) and provide them with a common `main()` that does some common setup (e.g., UART configs) and let them implement their own `test_main()` which `main()` will jump to.

Put another way, if a test developer included the OTTF ISRs lib (`sw_lib_testing_ottf_isrs`) in their test, but not the OTTF lib itself  (`sw_lib_testing_ottf`), then the kTestConfig won't be defined. So we don't want to check its first member because it won't exist. Does that make sense? This sacrifices a small bit of performance (2--4 instructions) to greatly improve maintainability.